adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,274 @@
+
+		requests = append(requests, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     a.endpointUri,
+			Body:    requestBody,
+			Headers: getHeaders(requestData),
+		})
+	}
+
+	return requests, errors
+}
+
+func processDataFromRequest(requestData *openrtb2.BidRequest, imp openrtb2.Imp, bidType openrtb_ext.BidType) resetDigitalRequest {
+
+	var resetDigitalRequestData resetDigitalRequest
+	resetDigitalRequestData.Site.Domain = requestData.Site.Domain
+	resetDigitalRequestData.Site.Referrer = requestData.Site.Page
+
+	resetDigitalRequestData.Imps = append(resetDigitalRequestData.Imps, resetDigitalRequesImps{})
+	resetDigitalRequestData.Imps[0].BidID = requestData.ID
+	resetDigitalRequestData.Imps[0].ImpID = imp.ID
+
+	var err error
+
+	if bidType == openrtb_ext.BidTypeBanner {
+		resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes = append(resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes, []int64{imp.Banner.Format[0].W, imp.Banner.Format[0].H})
+	}
+	if bidType == openrtb_ext.BidTypeVideo {
+		resetDigitalRequestData.Imps[0].MediaTypes.Video.Sizes = append(resetDigitalRequestData.Imps[0].MediaTypes.Banner.Sizes, []int64{*imp.Video.W, *imp.Video.H})
+	}
+
+	var extData = make(map[string]interface{})
+	err = json.Unmarshal(imp.Ext, &extData)
+	if err != nil {
+
+	} else {
+
+		resetDigitalRequestData.Imps[0].ZoneID.PlacementID = extData["bidder"].(map[string]interface{})["placement_id"].(string)
+		if resetDigitalRequestData.Imps[0].ZoneID.PlacementID == "test" {
+			resetDigitalRequestData.Imps[0].ForceBid = true
+		}
+
+	}
+	return resetDigitalRequestData
+
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, nil
+	}
+	if response.StatusCode != http.StatusOK {
+		return nil, nil
+	}
+
+	if err := json.Unmarshal(response.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(MaxBids)
+	//check no bids
+	jsonData := make(map[string]interface{})
+
+	json.Unmarshal([]byte(response.Body), &jsonData)
+	//Always one bid
+	bid := getBidFromResponse(jsonData)
+
+	bidType, err := getBidType(internalRequest.Imp[0])
+	if err != nil {
+		// handle error
+		return nil, []error{err}
+	}
+	bidResponse.Currency = getCurrency(internalRequest)
+	bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+		Bid:     bid,
+		BidType: bidType,
+	})
+
+	return bidResponse, nil
+}
+
+func getBidFromResponse(requestData map[string]interface{}) *openrtb2.Bid {
+	processData := requestData["bids"].([]interface{})[0].(map[string]interface{})
+
+	bid := &openrtb2.Bid{
+		ID:    processData["bid_id"].(string),
+		Price: getBidPrice(processData),
+		ImpID: processData["imp_id"].(string),
+		CrID:  processData["crid"].(string),
```

Comment:
these type casting may result in panic. consider unmarshalling `processData` into struct